### PR TITLE
disabled SegmentedControl options

### PR DESCRIPTION
**Pull request by petefine**
Fixes https://github.com/snehilvj/dash-mantine-components/issues/450 

As per base mantine, allow adding a disabled state to individual SegmentedControl items.

Enabled by setting `disabled` on a entry in the `data` property. 

I have zero typescript, javascript or react experience, so feel free to let me know if this is not correctly implemented. To test it, I used the following app:

```
import dash
import dash_mantine_components as dmc

from dash import Dash, _dash_renderer, Input, Output, dcc, html

_dash_renderer._set_react_version("18.2.0")

app = Dash(external_stylesheets=dmc.styles.ALL)

app.layout = dmc.MantineProvider(
    [
        dmc.SegmentedControl(
            id="segmented",
            data=[],
        ),
        html.Div("disable options:"),
        dmc.SegmentedControl(
            id="disabler",
            data=[
                "none",
                "a",
                "c",
                "ab",
            ],
            value="none",
        ),
    ]
)


@app.callback(
    Output("segmented", "data"),
    Input("disabler", "value"),
)
def update_disabled(choice):
    options = [
        {"label": "a", "value": "a"},
        {"label": "b", "value": "b"},
        {"label": "c", "value": "c"},
        {"label": "d", "value": "d"},
    ]
    if choice == "none":
        return options
    else:
        for option in options:
            if option["value"] in choice:
                option["disabled"] = True
    return options


if __name__ == "__main__":
    debug = True
    app.run_server(host="0.0.0.0", debug=debug, dev_tools_props_check=True)

```

If this work is accepted, I am happy to update the docs repository with a similar example if required.

### CODE REVIEW

**Comment by AnnMarieW** [comment]
Hi @petefine 

Thanks for reporting the issue and the PR!  :trophy: :

Your fix works perfectly.  Nice sample app too.  Here it is hosted on [PyCafe](https://py.cafe/snippet/dash/v1?#c=H4sIAODyVmcEA8VVTW_jNhD9K4J6SQBbtqTEtQ0Q2KJ7KbBFC7S3KBBoiY4Y06RKUt5VFvvf-4aSE3mTAHtbGrDNmTcfb2Y0-hpXphbxNpbH1lgf1dw1hZ5cyiPXXmpRVgZCLbR3EXdRfawKXei9NccAi0aTj_g_i8pgaYWuhRV2Fv2h287Por86H37rqppFjT8qcnGJTUonPG688uVJWCeNviridJ1kybKIr8mAt23EQqAr8cULq7kqne-VcI0Q3jGklgz35LdPn84mieK96TwsSf_nQOpva04Sca8KHeHcDT90CPSPeDiCr6h_N9pbo0bU-ciaFbE7Y4p4dqmuuefs7n4ivZ78J_bJR3kCuVo6vlMiMq0HW7cFzQnwRxMZvdh38riU0SlijXa-gtMpYv6OvHpHznevFFPmdE5cdQKJvop6Zntf6NCsQn-gflVcqR2vDiPZYXZQrmnFEZnovVQsDNpLTakawITQAwgRarGPuhZ2ohxh9VXVGFmJ6-3gZewEZmVSt69FrPhOqCLeEuELz6Pk25gFnUs4lecSDsn7cCrzJRyS9-FjKSZwSM5wlJV-5D4aSEaMATB0YeRLxwrfWX2mPsiFcmIC2Rs76iP5jJzo6SDMoLh7zuee0EPs78B0ntHnXpABi_61MA3g7xOjD6KUpeZHUZYDnRJrSuqyfKZUi133cOGHZsp2GvvFYrFcNcZ5TOMyCR8qYDBh4Zsup9Ibo1zZWtO6smpEdWDkDRMUz2Ir_uukFTSJDtuTdhgmizbZG_vyAx5337rtYtH2mOu9WDw0C2693GPNLZwWjVSnxwWZz0fz-Yv5Iss2-c1yk67yAHkjwnyZpLdJOm_7fE6dnXPdJ58bhUR939J6J0Pc2t43RuPe9qbG5puflkm2SlKoVNiP8VZ3Ss3ivcT-jLfYX7Ey1YGutHwQDMa_RA3cbaN9ttqn2S5bVfVyvb65zWuxyW7Wm92yus3W67zQOyX1QVjG0mSDKutKEGvJWLbMbrDTM4gabrHzkbc9ciWfCJ0neVApiaKzdZImvw7VhWFCL4PhhiLZaaVIS80clLRi31L-pF6MWXkacsZuh1z2ijswzHHLMda15nQB2eF1quRufhSe17TDUQdYQeVdzfWDsKYLpOjNqB-lfuQZGafk6cjtoWsd6BEiTW4LrYXzc-56XUlD_ViRWauMVz2lg3akuI9jQTViDLNBQpp1GIdgeUpmeCZtL_UD-cmTm0KjgV0bnhfGVhvkCTMnv5A-DYEwObySHqEwB0HQt3Awpzc4kHiyGUMGwXtnFYjnAzc4-izs4Ung2Qx1WhX6SbbIDiMS4M6htvG3_wGLX5U2yggAAA)


To help chip away at issues #421, could you please update the docstring on [line 49](https://github.com/snehilvj/dash-mantine-components/blob/master/src/ts/components/core/SegmentedControl.tsx#L49) so it has a better description?  The Mantine docs describe it as "A linear set of two or more segments", but that's not a great description either.  Maybe something like "SegmentedControl  is a horizontal selector for choosing one option from multiple segments"

Also to help with issue #364, could you please add this in the `/tests` folder ?

`test_segmented_control.py`

```python
from dash import Dash, html, Output, Input, _dash_renderer
import dash_mantine_components as dmc

_dash_renderer._set_react_version("18.2.0")


def test_001se_segmented_control(dash_duo):
    app=Dash()
    app.layout = dmc.MantineProvider(
        [
            dmc.SegmentedControl(
                id="segmented",
                data=[
                    {"label": "a", "value": "a"},
                    {"label": "b", "value": "b", "disabled": True},
                    {"label": "c", "value": "c"},
                ],
                value="a"
            ),
            html.Div(id="output"),
        ]
    )

    @app.callback(
        Output("output", "children"),
        Input("segmented", "value"),
    )
    def update(choice):
        return f"{choice=}"

    dash_duo.start_server(app)


    # Wait for the app to load
    dash_duo.wait_for_text_to_equal("#output", "choice='a'")

    option_b = dash_duo.find_element("input[value='b']")

    # Verify that "b" is disabled
    assert option_b.get_attribute("disabled") == 'true'

    option_c = dash_duo.find_element("input[value='c']").find_element_by_xpath(
        "./.."
    )
    option_c.click()
    dash_duo.wait_for_text_to_equal("#output", "choice='c'")

    assert dash_duo.get_logs() == []



```

**Comment by AnnMarieW** [comment]
Awesome!  
Let's see if Alex has any comments before merging, but it LGTM :slightly_smiling_face:

**Comment by petefine** [comment]
Great, thanks, having this will be very useful for my project! I have copied the maitine docs for this feature: https://github.com/snehilvj/dmc-docs/pull/121

**Review by alexcjohnson — APPROVED** [review]
💃 Looks great, and thanks for including a test!

**Comment by AnnMarieW** [comment]
@petefine - Congrats on your first PR in dash-mantine-components 🏆 
Thanks for adding this feature

**Pull request merged**
